pkg/controller/buildrun/runtime_image.go

```diff
@@ -0,0 +1,205 @@
+package buildrun
+
+import (
+	"bytes"
+	"fmt"
+	"path"
+	"strconv"
+	"strings"
+	"text/template"
+
+	buildv1alpha1 "github.com/redhat-developer/build/pkg/apis/build/v1alpha1"
+	v1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	// runtimeDockerfileTmpl Dockerfile template to be used with runtime-image, it uses Build
+	// attributes directly as template input.
+	runtimeDockerfileTmpl = `FROM {{ .Spec.Output.ImageURL }} as builder
+
+FROM {{ .Spec.Runtime.Base.ImageURL }}
+
+{{- range $k, $v := .Spec.Runtime.Env }}
+ENV {{ $k }}="{{ $v }}"
+{{- end }}
+
+{{- range $k, $v := .Spec.Runtime.Labels }}
+LABEL {{ $k }}="{{ $v }}"
+{{- end }}
+
+{{- range $cmd := .Spec.Runtime.Run }}
+RUN {{ $cmd }}
+{{- end }}
```

Comment:
This will cause many layers. Should we maybe generate this as:

```Dockerfile
RUN \
  cmd1 && \
  cmd2 && \
  cmd3
```

Reply:
@SaschaSchwarze0 linking commands with `&&` is already possible, by:

```yml
---
spec:
  runtime:
    run:
      - |
        cmd1 && \
        cmd2 && \
        cmd3
```

Which would then produce a single `RUN` directive in `Dockerfile.runtime`.